README.md

```diff
@@ -543,3 +572,45 @@ Nexus instead. The modified files will be accessible at the
 again in a future request, it will use it directly from Nexus rather than downloading it and
 uploading it again. This guarantees that any dependency used for a Cachito request can be used again
 in a future Cachito request.
+
+### pip
+
+The pip package manager works by parsing the `requirements.txt` and `requirements-build.txt` files
+present in the source repository to determine what dependencies are required to build the
+application. It is possible to specify different file path(s) for the requirements files as long
+as the files use the expected format.
+
+Cachito then creates two repositories in an instance of Nexus it manages that contain just the
+dependencies discovered in the requirements files. PyPI dependencies are uploaded to a PyPI hosted
+repository, external dependencies are uploaded to a raw repository. Connection information for the
+hosted repository is provided as the `PIP_INDEX_URL` environment variable accessible at the
+`/api/v1/requests/<id>/environment-variables` endpoint. To make external dependencies available,
+Cachito modifies the requirements files for the request by replacing relevant entries with their
+corresponding URLs from the raw repository. The modified requirements files are accessible at the
+`/api/v1/requests/<id>/configuration-files` endpoint.
+
+Cachito will produce a bundle that is downloadable at `/api/v1/requests/<id>/download`. This
+bundle will contain the application source code in the `app` directory and individual source
+archives of all the dependencies in the `deps/pip` directory. These archives are not meant to be
+used to build the application. They are there for convenience so that the dependency sources can be
+published alongside your application sources.
+
+In addition, they can be used to populate a local PyPI repository in the event that the application
+needs to be built without Cachito and the Nexus instance it manages. (TODO: does this apply?)
+
+As mentioned above, Cachito can also handle dependencies that are not from PyPI, such as those from
+a Git repository or an HTTP(S) URL. After downloading such a dependency, Cachito will upload it to
+the Nexus instance used for hosting permanent content. If Cachito encounters this same dependency
+again in a future request, it will use it directly from Nexus rather than downloading it and
+uploading it again. This guarantees that any dependency used for a Cachito request can be used again
+in a future Cachito request.
+
+Please note that if a dependency is from a private repository, set the
+[.netrc](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html) and
+`known_hosts` files for the Cachito workers. If the dependency location is not supported, Cachito
+will fail the request. (TODO: does this apply?)
```

Comment:
AFAIK this applies to npm only?

Reply:
Removed this, if it ends up being  relevant we can always copy it from the npm section